CMakeLists.txt

```diff
@@ -325,9 +325,15 @@ if(AVIF_ENABLE_NODISCARD)
     else()
         unset(CMAKE_C_STANDARD)
         set_property(TARGET avif_obj PROPERTY C_STANDARD)
-        target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:-std=gnu2x>>)
+        if(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
+            target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:-std=gnu2x>>)
+        elseif(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+            target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:/std:clatest>>)
+        endif()
+    endif()
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
+        target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:-Wunused-result>)
```

Comment:
[GCC documentation](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result) says:

> `-Wno-unused-result`
> Do not warn if a caller of a function marked with attribute `warn_unused_result` (see [Declaring Attributes of Functions](https://gcc.gnu.org/onlinedocs/gcc/Function-Attributes.html)) does not use its return value. The default is `-Wunused-result`.

[Clang documentation](https://clang.llvm.org/docs/DiagnosticsReference.html#wunused-result) says:

> **-Wunused-result**
> 
> This diagnostic is enabled by default.

Since the default is `-Wunused-result`, can we delete this?

Reply:
That sounds fine to me. It's safe to leave, but I doubt the default would be changed.

Reply:
yes, please delete

Reply:
Done.